### ISSUE TITLE
ThreadPool blocks application after closing

### ISSUE THREAD

**Issue by raphaelm** [edvin/tornadofx#585]
Hi,

first of all thank you for tornadoFX, I like it a lot! My application uses ``runAsync{ }`` in quite some places as the documentation recommends and I was having trouble with the fact that my application process continues to run for exactly 60 seconds, no matter what the async tasks were doing (even if they were empty). As the thread pool's queue was empty, I wonder why it still blocks shutdown.

Looking around the source code, I found out that it helps to either

* use ``runAsync(true)`` to use daemon threads
* add a ``currentStage?.setOnCloseRequest { terminateAsyncExecutors(5000) }`` to my main view

I'm not sure which of the two is the more graceful way to end them, so I'd kindly request for you to

* mention one of these two methods in the documentation
* or make it behave differently by default

Thank you!
Raphael

**Comment by edvin** [comment]
Hi Raphael,

Even if you're using the non-daemon thread pool, the application should quit on exit, provided that there are no live threads in the pool. I just confirmed that this is working as intended, so perhaps you can provide a small sample application that showcases the issue you're experiencing?

Please also check that you're running against 1.7.14 or 1.7.5-SNAPSHOT, we did some adjustments to this behavior a while back.

**Comment by raphaelm** [comment]
> so perhaps you can provide a small sample application that showcases the issue you're experiencing?

Yes, I did a small change to tornadofx-samples to showcase it: https://github.com/edvin/tornadofx-samples/compare/master...raphaelm:test-runasync

The output is 
```
Application started: 1515321392880
Running on thread pool: 1515321393436
Running on UI thread: 1515321393844
Closing window: 1515321394461
Application closed: 1515321394464
Shutdown hook: 1515321453438    <--- way after application closed!
```

However, apparently my application was running 1.7.12 (just as the samples to as well) and 1.7.14 fixes the problem already. Sorry for the noise!

**Comment by edvin** [comment]
Perfect! Good to have this documented and searchable. Thanks for reporting!
